vcd/resource_vcd_vapp_vm.go

```diff
@@ -96,18 +108,80 @@ func resourceVcdVAppVm() *schema.Resource {
 				Default:  true,
 			},
 			"network_href": &schema.Schema{
-				Type:     schema.TypeString,
-				Optional: true,
+				Type:       schema.TypeString,
+				Optional:   true,
+				Deprecated: "In favor of networks",
+			},
+			"networks": {
+				ConflictsWith: []string{"ip", "network_name", "vapp_network_name"},
+				ForceNew:      true,
+				Optional:      true,
+				Type:          schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"network_type": {
+							ForceNew:     true,
+							Required:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{"vapp", "org", "none"}, false),
+							Description:  "Network type to use: 'vapp', 'org' or 'none'. Use 'vapp' for vApp network, 'org' to attach Org VDC network. 'none' for empty NIC.",
+						},
+						"ip_allocation_mode": {
+							ForceNew:     true,
+							Optional:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{"POOL", "DHCP", "MANUAL", "NONE"}, false),
+						},
+						"network_name": {
+							ForceNew: false,
+							Optional: true, // In case of network_type = none it is not required
+							Type:     schema.TypeString,
+						},
+						"ip": {
+							Computed:     true,
+							ForceNew:     true,
+							Optional:     true,
+							Type:         schema.TypeString,
+							ValidateFunc: checkEmptyOrSingleIP(), // Must accept empty string to ease using HCL interpolation
+						},
+						"is_primary": {
+							Default:  false,
+							ForceNew: true,
+							Optional: true,
+							// schema change to false would not do anything useful unless
```

Comment:
Couldn't follow the comment, some details missing I guess ^^

Reply:
I hope it is better now.